go/controller/api/v1alpha1/autogenmodelconfig_types.go

```diff
@@ -24,11 +24,131 @@ const (
+	// Configuration for Anthropic provider models
+	// +optional
+	Anthropic *AnthropicConfig `json:"anthropic,omitempty"`
+
+	// Configuration for OpenAI provider models
+	// +optional
+	OpenAI *OpenAIConfig `json:"openAI,omitempty"`
+
+	// Configuration for Azure OpenAI provider models
+	// +optional
+	AzureOpenAI *AzureOpenAIConfig `json:"azureOpenAI,omitempty"`
+}
+
+// AnthropicConfig contains Anthropic-specific configuration options
+type AnthropicConfig struct {
+	// Base URL for the Anthropic API (overrides default)
+	// +optional
+	BaseURL string `json:"baseUrl,omitempty"`
+
+	// Maximum tokens to generate
+	// +optional
+	MaxTokens int `json:"maxTokens,omitempty"`
+
+	// Temperature for sampling
+	// +optional
+	Temperature string `json:"temperature,omitempty"`
+
+	// Top-p sampling parameter
+	// +optional
+	TopP string `json:"topP,omitempty"`
+
+	// Top-k sampling parameter
+	// +optional
+	TopK int `json:"topK,omitempty"`
+}
+
+// OpenAIConfig contains OpenAI-specific configuration options
+type OpenAIConfig struct {
+	// Base URL for the OpenAI API (overrides default)
+	// +optional
+	BaseURL string `json:"baseUrl,omitempty"`
+
+	// Organization ID for the OpenAI API
+	// +optional
+	Organization string `json:"organization,omitempty"`
+
+	// Temperature for sampling
+	// +optional
+	Temperature string `json:"temperature,omitempty"`
+
+	// Maximum tokens to generate
+	// +optional
+	MaxTokens int `json:"maxTokens,omitempty"`
+
+	// Top-p sampling parameter
+	// +optional
+	TopP string `json:"topP,omitempty"`
+
+	// Frequency penalty
+	// +optional
+	FrequencyPenalty string `json:"frequencyPenalty,omitempty"`
+
+	// Presence penalty
+	// +optional
+	PresencePenalty string `json:"presencePenalty,omitempty"`
+}
+
+// AzureOpenAIConfig contains Azure OpenAI-specific configuration options
+type AzureOpenAIConfig struct {
+	// Endpoint for the Azure OpenAI API
+	// +optional
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// API version for the Azure OpenAI API
+	// +optional
+	APIVersion string `json:"apiVersion,omitempty"`
+
+	// Deployment name for the Azure OpenAI API
+	// +optional
+	DeploymentName string `json:"deploymentName,omitempty"`
+
+	// Temperature for sampling
+	// +optional
+	Temperature string `json:"temperature,omitempty"`
+
+	// Maximum tokens to generate
+	// +optional
+	MaxTokens int `json:"maxTokens,omitempty"`
+
+	// Top-p sampling parameter
+	// +optional
+	TopP string `json:"topP,omitempty"`
+}
+
 // ModelConfigSpec defines the desired state of ModelConfig.
 type ModelConfigSpec struct {
-	Model            string `json:"model"`
+	Model string `json:"model"`
+
+	// The provider of the model
+	// +kubebuilder:default=OpenAI
+	Provider ModelProvider `json:"provider"`
+
 	APIKeySecretName string `json:"apiKeySecretName"`
 	APIKeySecretKey  string `json:"apiKeySecretKey"`
+
+	// Provider-specific configuration
+	// +optional
+	ProviderConfig *ProviderConfig `json:"providerConfig,omitempty"`
```

Comment:
I think the fields in here should be flattened into the `ModelConfigSpec`. The Gateway API does validation via `cel` to ensure that the `ModelProvider` matches the field chosen. It's a nice UX.

Reply:
I want to bring this comment back up, I think it's important

Reply:
yep, still working on this